### PR TITLE
Fix/link from fees

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
https://reload.atlassian.net/browse/DDFLSBP-553

#### Description
If the user has overdue loans, a warning bar pops up on the fees page. This warning bar has a CTA button that was linking to the loans page. Now it opens the modal with overdue loans as the client asked for.

#### Screenshot of the result
-

#### Additional comments or questions
-

### CODE REVIEW

File: src/core/utils/helpers/modal-helpers.ts [hunk 1]
```diff
@@ -45,3 +45,9 @@ export const getDetailsModalId = (queryParam: string, prefix: string) => {
   }
   return "";
 };
+
+export const createLoanModalId = (
+  dueDate: string | null | undefined,
+  dueDateModal: string | number | Record<string, unknown>,
+  allLoansId: string | number | Record<string, unknown>
+) => (dueDate ? `${dueDateModal}-${dueDate}` : String(allLoansId));
```
Comment (spaceo) [inline src/core/utils/helpers/modal-helpers.ts:53]: I think we should define modalId functions in the modal components.
That makes it more modular. If we want to delete the modal in the future the whole thing is self-conatined.
Plus that if there is some functionality involved in finding out the omdal id fragments, it probably comes from functionality inside of the component itself.

Comment (spaceo) [reply src/core/utils/helpers/modal-helpers.ts:53]: For consistency I think you should use this function when creating the id:
https://github.com/danskernesdigitalebibliotek/dpl-react/blob/4f75d55e348f0395031fa4d8d76397ece159e504/src/core/utils/helpers/modal-helpers.ts#L3-L5

File: src/apps/fee-list/FeeList.entry.tsx [hunk 2]
```diff
@@ -62,8 +67,21 @@ const FeeListEntry: FC<
...
+    RenewalArgs
+> = ({ pageSizeDesktop, pageSizeMobile }) => {
+  const pageSize = pageSizeGlobal(
+    {
+      desktop: pageSizeDesktop,
+      mobile: pageSizeMobile
+    },
+    "pageSizeLoanList"
+  );
+  return <FeeList pageSize={pageSize} />;
```
Comment (spaceo) [inline src/apps/fee-list/FeeList.entry.tsx:83]: Would be nice with a comment here.
You are writing `pageSizeLoanList ` although we are dealing with a fee list?

**Review by spaceo — CHANGES_REQUESTED** [review]
Nice work. I have some comments.

File: src/core/utils/helpers/modal-helpers.ts (hunk 1 above)
Comment (Adamik10) [reply src/core/utils/helpers/modal-helpers.ts:53]: @spaceo but we've been creating this specific modal id differently. I just took the function out into a helper.

Comment (spaceo) [reply src/core/utils/helpers/modal-helpers.ts:53]: I don't think you need a full overhaul. Just like it has been done eg. here:
https://github.com/danskernesdigitalebibliotek/dpl-react/blob/4f75d55e348f0395031fa4d8d76397ece159e504/src/components/find-on-shelf/FindOnShelfModal.tsx#L12-L14

**Review by spaceo — CHANGES_REQUESTED** [review]
Comment back

File: src/core/utils/helpers/modal-helpers.ts (hunk 1 above)
Comment (Adamik10) [reply src/core/utils/helpers/modal-helpers.ts]: donezies :)

In regards to the self-containment of modal related stuff - I think it's pretty self contained when the function exists in a file called `modal-helpers.ts`

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**